### DISCUSSION TITLE
Autosave for new files?

### DISCUSSION THREAD

**Discussion by zack-kimble** [Zettlr/Zettlr#3336] [Q&A]
I use Zettlr largely for note taking, which means I often create new files and don't immediately rename them. In the past, I thought these would autosave with the default file name, though perhaps I'm mistaken.

Yesterday I took notes in a new file like this an then later closed the app. This file appears to have disappeared. I have tested and if I open a new file it maintains an '*' by the file name for far longer than 5 seconds, even when new changes are made. Additionally, if I hit `crtl-s` Zettlr says the save file is ready, but no file appears to be saved. (Edit: apparently, this was opening up a save dialogue window that Ubuntu was placing behind the main Zettlr window). 

Is it a bug that new files are not autosaved or did something change about how autosave works for default file names?

**Comment by nathanlesage** [comment]
Here's a relevant link to this problem: https://github.com/Zettlr/Zettlr/issues/2480

**Comment by mjatti** [comment]
Hijacking this old thread.

Installed 4.20 today and started a new file, say FILE.md. Seven hours later my Macbook (Tahoe 26.2) suddenly logged me out for reasons unknown. Logging back the OS automatically opened Zettlr, only for me to see that FILE.md was empty and I couldn't bring the text back.

Checking the settings, autosave feature is disabled by default. Fair enough, but I also feel that editors such as VS Code handle similar situations more gracefully. I would expect the editor to save the last buffer and show it after a restart for whatever reason.

For those arguing "WhY dIDnT JOO SaVe JooR FILE?!": yes, it would've possibly helped me personally somewhat, depending on when I last saved the file. However, this doesn't address the actual issue: I would expect a "batteries included" editor such as Zettlr to a) keep the last state and b) possibly also have autosave enabled by default.

I did use Time Machine to see ~/Library/Application Support/Zettlr history, but couldn't immediately figure out where a possible cached version of my work is. If someone has any idea if such a buffer is cached, let me know. FILE.md was 0 bytes in all snapshots as expected when autosave is disabled.

**Reply by nathanlesage** [reply]
Zettlr currently doesn't have such a feature, but the development team behind it is also much smaller than that of VS Code. We're slowly working our way through the batteries to get them included, so to speak. Data loss is never great, I fully agree, and that's why those things are always a priority.
